src/tests/unit/tests/background/actions/details-view-action-creator.test.ts

```diff
@@ -0,0 +1,120 @@
+
+        detailsViewActionsMock.setup(actions => actions['openSettingsPanel']).returns(() => openSettingsPanelMock.object);
+
+        const payload: BaseActionPayload = {
+            telemetry: {
+                triggeredBy: 'test' as TriggeredBy,
+                source: -1 as TelemetryEventSource,
+            },
+        };
+
+        const tabId = -1;
+
+        registerCallbackMock
+            .setup(register => register(Messages.SettingsPanel.OpenPanel, It.is(isFunction)))
+            .callback((message, listener) => listener(payload, tabId));
+
+        testSubject.registerCallback();
+
+        openSettingsPanelMock.verify(action => action.invoke(null), Times.once());
+        detailsViewControllerMock.verify(controller => controller.showDetailsView(tabId), Times.once());
+        telemetryEventHandlerMock.verify(handler => handler.publishTelemetry(SETTINGS_PANEL_OPEN, payload), Times.once());
+    });
+
+    test('on SettingsPanel.ClosePanel', () => {
+        const closeSeetingsPanelMock = Mock.ofType<Action<null>>();
+
+        detailsViewActionsMock.setup(actions => actions['closeSettingsPanel']).returns(() => closeSeetingsPanelMock.object);
+
+        const payload: BaseActionPayload = {
+            telemetry: {
+                triggeredBy: 'test' as TriggeredBy,
+                source: -1 as TelemetryEventSource,
+            },
+        };
+
+        const tabId = -1;
+
+        registerCallbackMock
+            .setup(register => register(Messages.SettingsPanel.ClosePanel, It.is(isFunction)))
+            .callback((message, listener) => listener(payload, tabId));
+
+        testSubject.registerCallback();
+
+        closeSeetingsPanelMock.verify(action => action.invoke(null), Times.once());
+        telemetryEventHandlerMock.verify(handler => handler.publishTelemetry(SETTINGS_PANEL_CLOSE, payload), Times.once());
+    });
+
+    test('on Visualization.DetailsView.SetDetailsViewRightContentPanel', () => {
+        const setSelectedPanelMock = Mock.ofType<Action<DetailsViewRightContentPanelType>>();
+
+        detailsViewActionsMock
+            .setup(actions => actions['setSelectedDetailsViewRightContentPanel'])
+            .returns(() => setSelectedPanelMock.object);
+
+        const payload: DetailsViewRightContentPanelType = 'Overview';
+
+        registerCallbackMock
+            .setup(register => register(Messages.Visualizations.DetailsView.SetDetailsViewRightContentPanel, It.is(isFunction)))
+            .callback((message, listener) => listener(payload));
+
+        testSubject.registerCallback();
+
+        setSelectedPanelMock.verify(action => action.invoke(payload), Times.once());
+    });
+
+    test('on Visualization.DetailsView.GetState', () => {
+        const getCurrentStateMock = Mock.ofType<Action<null>>();
+
+        detailsViewActionsMock.setup(actions => actions['getCurrentState']).returns(() => getCurrentStateMock.object);
+
+        registerCallbackMock
```

Comment:
Might not be a bad idea to move this to a private method; seems duplicated in this test file.

Reply:
the setting of the `registerCallbackMock`?? we have 3 different ways of calling the listener (no params, 1 and 2 params). Tests are short enough that I don't think this is a big deal. I'm planning on coming back to the action creator tests (once I finish breaking up the current action creator and global action creator) and apply something similar to what we have for the original action creator, i.e. an action creator tester class. But that seems to be out of scope for this PR.